File: QuanLyPhongKham.Server/src/QuanLyPhongKham.Models/Entities/BENHNHAN.cs
using MayNghien.Infrastructures.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Reflection.Metadata.Ecma335;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyPhongKham.Models.Entities
{
    public class BENHNHAN : BaseEntity
    {
        [Key]
        public Guid MaBN { get; set; }
        public string HoTen { get; set; }
        public DateTime NgaySinh { get; set; }
        public bool GioiTinh { get; set; }
        public string? SoDienThoai { get; set; }
        public string? DiaChi { get; set; }
        public string? SoBHYT { get; set; }
        public string? TienSuDiUng { get; set; }

        public ICollection<LICHHEN> LichHen { get; set; }
    }
}

File: QuanLyPhongKham.Client/src/models/BenhNhan.js
/**
 * @typedef {Object} BenhNhan
 * @property {string} maBN
 * @property {string} hoTen
 * @property {string|Date} ngaySinh
 * @property {boolean} gioiTinh
 * @property {string} soDienThoai
 * @property {string} diaChi
 * @property {string} soBHYT
 * @property {string} tienSuDiUng
 */

export const createBenhNhan = (
  maBN = "",
  hoTen = "",
  ngaySinh = null,
  gioiTinh = false,
  soDienThoai = "",
  diaChi = "",
  soBHYT = "",
  tienSuDiUng = "",
) => ({
  maBN,
  hoTen,
  ngaySinh,
  gioiTinh,
  soDienThoai,
  diaChi,
  soBHYT,
  tienSuDiUng,
});

export const normalizeBenhNhan = (item) => ({
  maBN: item?.maBN ?? item?.MaBN ?? "",
  hoTen: item?.hoTen ?? item?.HoTen ?? "",
  ngaySinh: item?.ngaySinh ?? item?.NgaySinh ?? null,
  gioiTinh: item?.gioiTinh ?? item?.GioiTinh ?? false,
  soDienThoai: item?.soDienThoai ?? item?.SoDienThoai ?? "",
  diaChi: item?.diaChi ?? item?.DiaChi ?? "",
  soBHYT: item?.soBHYT ?? item?.SoBHYT ?? "",
  tienSuDiUng: item?.tienSuDiUng ?? item?.TienSuDiUng ?? "",
});
